scaleway/resource_mnq_queue.go

```diff
@@ -0,0 +1,421 @@
+				ValidateFunc: validation.IntBetween(0, 43_200),
+			},
+			"url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL of the queue",
+			},
+		},
+	}
+}
+
+func resourceScalewayMNQQueueCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := newMNQAPI(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	namespaceRegion, namespaceID, err := parseRegionalID(d.Get("namespace_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if region != namespaceRegion {
+		return diag.Errorf("region of the queue (%s) and the namespace (%s) must match", region, namespaceRegion)
+	}
+
+	namespace, err := api.GetNamespace(&mnq.GetNamespaceRequest{
+		Region:      namespaceRegion,
+		NamespaceID: namespaceID,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	switch namespace.Protocol {
+	case mnq.NamespaceProtocolSqsSns:
+		return resourceScalewayMNQQueueCreateSQS(ctx, d, meta)
+	case mnq.NamespaceProtocolNats:
+		return resourceScalewayMNQQueueCreateNATS(ctx, d, meta)
+	default:
+		return diag.Errorf("unknown protocol %s", namespace.Protocol)
+	}
+}
+
+func resourceScalewayMNQQueueCreateSQS(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, _, err := SQSClientWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	isFifo := d.Get("fifo_queue").(bool)
+	sqsConfig := d.Get("sqs")
+
+	attributes := SQSResourceToAttributes(resourceScalewayMNQQueueSQS().Schema, sqsConfig.([]interface{})[0].(map[string]interface{}))
+	attributes[sqs.QueueAttributeNameFifoQueue] = aws.String(strconv.FormatBool(isFifo))
+
+	name := resourceMNQQueueName(d.Get("name"), d.Get("name_prefix"), true, isFifo)
+	input := &sqs.CreateQueueInput{
+		QueueName:  aws.String(name),
+		Attributes: attributes,
+	}
+
+	_, err = retryWhenAWSErrCodeEquals(ctx, []string{sqs.ErrCodeQueueDeletedRecently}, &RetryWhenConfig[*sqs.CreateQueueOutput]{
+		Timeout:  d.Timeout(schema.TimeoutCreate),
+		Interval: DefaultMNQQueueRetryInterval,
+		Function: func() (*sqs.CreateQueueOutput, error) {
+			return api.CreateQueueWithContext(ctx, input)
+		},
+	})
+	if err != nil {
+		return diag.Errorf("failed to create SQS Queue: %s", err)
+	}
+
+	namespaceRegion, namespaceID, err := parseRegionalID(d.Get("namespace_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(composeMNQQueueID(namespaceRegion, namespaceID, *input.QueueName))
+
+	return resourceScalewayMNQQueueReadSQS(ctx, d, meta)
+}
+
+func resourceScalewayMNQQueueCreateNATS(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := resourceMNQQueueName(d.Get("name"), d.Get("name_prefix"), false, false)
+
+	namespaceRegion, namespaceID, err := parseRegionalID(d.Get("namespace_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(composeMNQQueueID(namespaceRegion, namespaceID, name))
+
```

Comment:
Should the creation be a jetstream ? It seems like the best take for what could be considered a queue